esmvalcore/cmor/check.py

```diff
@@ -91,23 +98,56 @@ def check_metadata(self, logger=None):
             all checks and then raises.
 
         """
-        if logger is None:
-            logger = logging.getLogger(__name__)
+        if logger is not None:
+            self._logger = logger
 
         self._check_var_metadata()
         self._check_fill_value()
         self._check_dim_names()
         self._check_coords()
         if self.frequency != 'fx':
             self._check_time_coord()
+
         self._check_rank()
 
-        self.report_debug_messages(logger)
-        self.report_warnings(logger)
+        self.report_debug_messages()
+        self.report_warnings()
         self.report_errors()
 
         if self.frequency != 'fx':
-            self._add_auxiliar_time_coordinates()
+            self._add_auxiliary_time_coordinates()
+
+        return self._cube
+
+    def check_data(self, logger=None):
+        """Check the cube data.
+
+        Performs all the tests that require to have the data in memory.
+        Assumes that metadata is correct, so you must call check_metadata prior
+        to this.
+
+        It will also report some warnings in case of minor errors.
+
+        Raises
+        ------
+        CMORCheckError
+            If errors are found. If fail_on_error attribute is set to True,
+            raises as soon as an error is detected. If set to False, it perform
+            all checks and then raises.
+
+        """
+        if logger is not None:
+            self._logger = logger
+
+        if self._cmor_var.units:
+            units = self._get_effective_units()
+            if str(self._cube.units) != units:
+                self._cube.convert_units(units)
```

Comment:
I would expect some sort of warning when the data has the wrong units?

Reply:
This comment has been addressed for coordinate units, but not for the units of the variable. Could you have a look please?

Reply:
It's being reported while checking, in line 264. But I can add the warning here if you prefer.